src/core/biblio.js

```diff
@@ -93,24 +92,78 @@ export function wireReference(rawRef, target = "_blank") {
   `;
 }
 
-export function stringifyReference(ref) {
-  if (typeof ref === "string") return ref;
-  let output = `<cite>${ref.title}</cite>`;
-  if (ref.href) {
-    output = `<a href="${ref.href}">${output}</a>. `;
+// Author, A. (Year, Month Date of Publication). Article title. Retrieved from URL
+function entryToAPA(ref) {
+  const authors =
+    ref.authors && ref.authors.length
+      ? `${ref.authors.join("; ")} ${ref.etAl ? " et al" : ""}.`
+      : "";
+  const date = ref.date ? `(${ref.date}).` : "";
+  const title = ref.href
+    ? `<a href="${ref.href}"><cite>${ref.title}</cite></a>. `
+    : "";
+  const URL = ref.href
+    ? `Retrieved from URL: <a href="${ref.href}">${ref.href}</a>`
+    : "";
+
+  return `${authors} ${date} ${title} ${URL}`;
+}
+
+//Author’s Last name, First name. “Title of the Article or Individual Page.” Title of the website, Name of the publisher, Date of publication, URL.
+function entryToMLA(ref) {
+  const authors = "";
+  if (ref.authors && ref.authors.length) {
+    authors = ref.authors.join("; ") + (ref.etAl ? " et al." : ".");
+  }
+  const date = ref.date ? `${ref.date}, ` : "";
+  const title = ref.href
+    ? `<a href="${ref.href}"> "<cite>${ref.title}</cite>" </a>. `
+    : "";
+  const URL = ref.href ? `<a href="${ref.href}">${ref.href}</a>` : "";
+  const publisher = "";
+  if (ref.publisher) {
+    publisher = ref.publisher + ref.publisher.endsWith(".");
   }
+
+  return `${authors} ${title} ${publisher} ${date} ${URL}`;
+}
+
+// what ReSpec currently does...
+function entryToW3C(ref) {
+  const title = ref.href
+    ? `<a href="${ref.href}"> <cite>${ref.title}</cite> </a>. `
+    : "";
+  const authors = "";
   if (ref.authors && ref.authors.length) {
-    output += ref.authors.join("; ");
-    if (ref.etAl) output += " et al";
-    output += ".";
+    authors = ref.authors.join("; ") + (ref.etAl ? " et al." : ".");
   }
+  const publisher = "";
   if (ref.publisher) {
-    const publisher = ref.publisher + (/\.$/.test(ref.publisher) ? "" : ".");
-    output = `${output} ${publisher} `;
+    publisher = ref.publisher + ref.publisher.endsWith(".");
```

Comment:
Same concatenation.

Reply:
Adding some tests should pick up any other bugs like this.